packages/search-ui-elasticsearch-connector/src/transporter/__tests__/ApiClientTransporter.test.ts

```diff
@@ -0,0 +1,177 @@
+import { ApiClientTransporter } from "../ApiClientTransporter";
+import type {
+  ConnectionOptions,
+  SearchRequest,
+  ResponseBody
+} from "../../types";
+import { getHostFromCloud } from "../../utils";
+
+jest.mock("../../utils", () => ({
+  getHostFromCloud: jest.fn()
+}));
+
+describe("ApiClientTransporter", () => {
+  const mockFetch = jest.fn();
+  global.fetch = mockFetch;
+
+  beforeEach(() => {
+    mockFetch.mockClear();
+    (getHostFromCloud as jest.Mock).mockClear();
+  });
+
+  const createTransporter = (config: Partial<ConnectionOptions> = {}) => {
+    return new ApiClientTransporter({
+      host: "http://localhost:9200",
+      index: "test-index",
+      ...config
+    });
+  };
+
+  const mockResponse: ResponseBody = {
+    took: 1,
+    timed_out: false,
+    _shards: {
+      total: 1,
+      successful: 1,
+      skipped: 0,
+      failed: 0
+    },
+    hits: {
+      total: { value: 1, relation: "eq" },
+      hits: []
+    }
+  };
+
+  it("should make a request with correct headers", async () => {
+    const transporter = createTransporter();
+    const searchRequest: SearchRequest = { query: { match_all: {} } };
+
+    mockFetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockResponse)
+    });
+
+    await transporter.performRequest(searchRequest);
+
+    expect(mockFetch).toHaveBeenCalledWith(
+      "http://localhost:9200/test-index/_search",
+      expect.objectContaining({
+        method: "POST",
+        headers: expect.objectContaining({
+          "Content-Type": "application/json",
+          "x-elastic-client-meta": expect.any(String)
+        }),
+        body: JSON.stringify(searchRequest)
+      })
+    );
+  });
+
+  it("should include api key in headers when provided", async () => {
+    const transporter = createTransporter({
+      apiKey: "test-api-key"
+    });
+
+    mockFetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockResponse)
+    });
+
+    await transporter.performRequest({ query: { match_all: {} } });
+
+    expect(mockFetch).toHaveBeenCalledWith(
+      expect.any(String),
+      expect.objectContaining({
+        headers: expect.objectContaining({
+          Authorization: "ApiKey test-api-key"
+        })
+      })
+    );
+  });
+
+  it("should use cloud host when cloud config is provided", async () => {
+    const cloudConfig = {
+      id: "test-cloud-id",
+      username: "test-user",
+      password: "test-password"
+    };
+
+    (getHostFromCloud as jest.Mock).mockReturnValue("https://cloud-host.com");
```

Comment:
we don't have any tests for `getHostFromCloud` so it could be good to not mock this an test with a stubbed cloud id here. and also write some unit tests for invalid cloud ids to make sure it errors in ways we expect. IE gives useful error messages if the cloud id fails to decode.

Reply:
Added utils tests for node and browser env